### PR TITLE
rustscan 1.3 (new formula)

### PR DESCRIPTION
**Pull request by bee-san**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/rustscan.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  license "MIT"
+  depends_on "rust" => :build
+  depends_on "nmap" => :optional
+  def install
+    system "cargo", "install", *std_cargo_args
+    # bin.install "target/release/rustscan"
+  end
+
+  test do
+    # Due to law, users installing this package may not be able to portscan websites
```
Comment (SMillerDev) [inline Formula/rustscan.rb:16]: Will it fail if you scan something like 0.0.0.0?

Comment (brandonskerritt) [reply Formula/rustscan.rb:16]: Nope! Do you want me to add that as a test? :)

Comment (SMillerDev) [reply Formula/rustscan.rb:16]: It's a result we can be pretty sure of. I think it would be a good fit.

File: Formula/rustscan.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  homepage "https://github.com/rustscan/rustscan"
+  url "https://github.com/RustScan/RustScan/archive/1.3.tar.gz"
+  version "1.3.0"
+  sha256 "3bbaf188fa4014a57596c4d4f928b75bdf42c058220424ae46b94f3a36b61f81"
+  license "MIT"
+  depends_on "rust" => :build
+  depends_on "nmap" => :optional
+  def install
+    system "cargo", "install", *std_cargo_args
+    # bin.install "target/release/rustscan"
```
Comment (dawidd6) [inline Formula/rustscan.rb:12]:
```suggestion
```

File: Formula/rustscan.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Rustscan < Formula
+  desc "Faster Nmap Scanning with Rust"
+  homepage "https://github.com/rustscan/rustscan"
+  url "https://github.com/RustScan/RustScan/archive/1.3.tar.gz"
+  version "1.3.0"
+  sha256 "3bbaf188fa4014a57596c4d4f928b75bdf42c058220424ae46b94f3a36b61f81"
+  license "MIT"
+  depends_on "rust" => :build
+  depends_on "nmap" => :optional
```
Comment (dawidd6) [inline Formula/rustscan.rb:9]:
```suggestion

  depends_on "nmap" => :optional
```

File: Formula/rustscan.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Rustscan < Formula
+  desc "Faster Nmap Scanning with Rust"
+  homepage "https://github.com/rustscan/rustscan"
+  url "https://github.com/RustScan/RustScan/archive/1.3.tar.gz"
+  version "1.3.0"
```
Comment (dawidd6) [inline Formula/rustscan.rb:5]:
```suggestion
```

No need for that, will be determined automatically.

File: Formula/rustscan.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "3bbaf188fa4014a57596c4d4f928b75bdf42c058220424ae46b94f3a36b61f81"
+  license "MIT"
+  depends_on "rust" => :build
```
Comment (dawidd6) [inline Formula/rustscan.rb:8]:
```suggestion

  depends_on "rust" => :build
```

File: Formula/rustscan.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "nmap" => :optional
+  def install
```
Comment (dawidd6) [inline Formula/rustscan.rb:10]:
```suggestion

  def install
```

File: Formula/rustscan.rb (hunk 3 above)
Comment (dawidd6) [inline Formula/rustscan.rb:9]: We don't support optional dependencies in core tap. Is `nmap` needed at runtime?

**Comment by chenrui333** [comment]
```
rustscan:
  * License mismatch - GitHub license is: ["GPL-3.0"], but Formulae license states: ["MIT"].
  * GitHub repository too new (<30 days old)
```

While we can get away with github repo is too new, but the license mismatch seems interesting.

File: Formula/rustscan.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+class Rustscan < Formula
+  desc "Faster Nmap Scanning with Rust"
+  homepage "https://github.com/rustscan/rustscan"
+  url "https://github.com/RustScan/RustScan/archive/1.3.tar.gz"
+  version "1.3.0"
+  sha256 "3bbaf188fa4014a57596c4d4f928b75bdf42c058220424ae46b94f3a36b61f81"
+  license "MIT"
```
Comment (chenrui333) [inline Formula/rustscan.rb:7]:
```suggestion
  license "GPL-3.0"
```

**Comment by SMillerDev** [comment]
> While we can get away with github repo is too new, but the license mismatch seems interesting.

The audit is there for a reason. We can reconsider when the software has had some time to mature. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “maturity” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**